### ISSUE TITLE
fix(agent): gemma4 partial prefix reuse prefills only the suffix — the "reuse is inert" invariant at gemma4_session.go:33 does not hold

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#5548]
## A documented invariant that does not hold

`internal/model/gemma4_session.go:30-34` states, as the reason the recompute bridge is safe:

> Because it holds no K/V rows it leaves s.Cache untouched: `Cache.Len()` stays 0, and
> **eviction / prefix reuse are INERT rather than silently wrong.**

Eviction is indeed inert. **Prefix reuse is not.** A gemma4 session served through
`InKernelPlanner` with a *partial* prefix hit prefills only the suffix, against a session that
carries none of the prefix's history — wrong logits, no error, no panic.

## The path, link by link

Each line below was read at HEAD (local trunk, 51 commits ahead of `origin/main`).

1. `internal/agent/inkernel_planner.go:210-215` — reuse is enabled unconditionally on the host
   path:
   ```go
   func inKernelPlannerPrefixReuseSupported(m *model.Model, backend compute.Backend) bool {
       if backend == nil {
           return true
       }
       return m != nil && m.Cfg.InKernelBackendPrefixReuseSupported()
   }
   ```
   `backend == nil` is *exactly* the mode the gemma4 bridge is wired for —
   `gemma4_session.go:47-49` requires `s.Backend == nil && !s.Metal && !s.MetalQ4K &&
   s.PrecisionPolicy == nil`. So the one configuration gemma4 runs in is the one that skips the
   capability check entirely.

2. `internal/agent/inkernel_decode.go:196` — after a turn, `snap := s.Cache.Clone()` is admitted
   into the radix tree keyed by the full `ids`. For gemma4 that snapshot is **empty**.

3. `internal/radixkv/radixkv.go:793-806` — splitting that node does not reject it:
   ```go
   func truncatePrefix(c *model.KVCache, L int) *model.KVCache {
       cp := c.Clone()
       if L < cp.Len() { ... }   // Len()==0 ⇒ L < 0 ⇒ false
       return cp                 // non-nil, zero-length
   }
   ```

4. `internal/agent/inkernel_decode.go:68` — the nil-guard is the only gate, and an empty cache is
   not nil: `if k := b.KV(); k != nil { matchedKV = k.Clone() }`. Tree matching is on **token
   ids**, not on KV depth, so a later prompt sharing a prefix returns `m > 0`.

5. `internal/agent/inkernel_decode.go:418-427` — the rebuilt session drops the history:
   ```go
   s := p.m.NewSession()
   s.Cache = prefix
   ```
   `gemma4Hist` is never copied. `model.SessionFromPrefix` (`internal/model/kv.go:14-16`) has the
   same hole.

6. `internal/agent/inkernel_decode.go:157,173` — `prefillAt := matched`, then
   `logits = s.Prefill(ids[prefillAt:])`. The gemma4 recompute forward runs over **the suffix
   alone**, having been told the prefix is already in cache.

**The exact-hit case accidentally fails safe** and is why this has not been noticed:
`inKernelRefeedLastTokenForExactHit` (`inkernel_decode.go:429-430`) sees
`s.Cache.Len() (0) < promptLen` and discards the session. Only the **partial** hit is wrong —
i.e. the second turn of any multi-turn session, which is the common case in serve.

## Red-first witness — the harness already exists

`internal/ggufload/gemma4_test.go:157 tinyGemma4GGUF` builds a faithful 4-layer gemma4 GGUF
(3 sliding + 1 global, `head_dim` 16/32, `kv_heads` 2/1, `sliding_window: 2`), and
`internal/ggufload/gemma4_session_test.go:33` already asserts Session logits are bit-exact with
`Model.Forward`. So the new test is a small delta on an existing pattern:

- prefill prompt A through the reuse path and admit it;
- serve prompt A+B sharing a ~60% prefix through the same planner;
- compare against `Model.Forward(A+B)`.

That is red today. Note a plain unit test on `Session` alone will NOT catch this — the defect
lives in the planner's reuse wiring, so the test has to go through the reuse path.

## Two candidate fixes

1. **Fail closed** — a `Config.isGemma4()` arm in `inKernelPlannerPrefixReuseSupported`. Smallest
   diff; makes the `gemma4_session.go:33` comment true.
2. **Carry the history** — teach `SessionFromPrefix` / the planner's prefix clone to copy
   `gemma4Hist`. Strictly better: reuse becomes *correct but not yet cheap* (the recompute still
   re-runs the whole prefix), which turns the eventual cached path (#5496) into a pure
   performance change rather than a semantics change.

Either way the false sentence at `gemma4_session.go:33` must be corrected, since it is the thing
that made the hazard invisible.

## Scope note

This is **not** #5496. #5496 asks for a cached KV path (a performance feature, and per-layer
window bounding there is genuinely epic-sized because `KVCache.pos` is one array shared across
layers — `internal/model/kvcache.go:17`). This ticket is a correctness bug that exists *because*
#5496 has not landed, and it should be fixed first and independently.

## Open question an implementer should settle first (~10 min)

Whether a gemma4 GGUF can currently boot end-to-end through `fak serve` on the `InKernelPlanner`
route. Every link above was verified by reading, but that last step was not. If serve cannot yet
reach it, this is a **latent** bug rather than a live one — still worth landing fail-closed, but
the urgency changes. Say which it is in the fix.

## Done when

- [ ] A test drives a partial prefix hit for gemma4 through the planner's reuse path and compares
      against `Model.Forward`, and it is red before the fix.
- [ ] Partial prefix reuse for gemma4 either fails closed or carries `gemma4Hist`.
- [ ] `internal/model/gemma4_session.go:33`'s "INERT rather than silently wrong" claim is true, or
      corrected.
